comb_spec_searcher/comb_spec_searcher.py

```diff
@@ -570,85 +560,69 @@ def auto_search(self, **kwargs) -> CombinatorialSpecification:
         tree that is as small as possible.
         """
         auto_search_start = time.time()
-
-        perc = kwargs.get("perc", 1)
-        if not 0 < perc <= 100:
-            logger.warning(
-                (
-                    "Percentage not between 0 and 100, so assuming 1%"
-                    " search percentage."
-                ),
-            )
-            perc = 1
-        status_update = kwargs.get("status_update", None)
-        max_time = kwargs.get("max_time", None)
-        status_start = time.time()
         start_string = "Auto search started {}\n".format(
             time.strftime("%a, %d %b %Y %H:%M:%S", time.gmtime())
         )
         start_string += self.run_information()
         logger.info(start_string)
-
-        max_expansion_time = 0
-        expanding = True
-
-        while expanding:
-            expanding, status_start = self._expand_classes_for(
-                max_expansion_time, status_update, status_start, auto_search_start
-            )
-
-            spec_search_start = time.time()
-            logger.debug("Searching for specification.")
-            specification = self.get_specification(
-                smallest=kwargs.get("smallest", False),
-                minimization_time_limit=0.01 * (time.time() - auto_search_start),
-            )
-            if specification is not None:
-                self._log_spec_found(specification, auto_search_start)
-                return specification
-            logger.debug("No specification found.")
-            if max_time is not None:
-                if time.time() - auto_search_start > max_time:
-                    raise ExceededMaxtimeError(
-                        "Exceeded maximum time. Aborting auto search.",
-                    )
-            # worst case, search every hour
-            multiplier = 100 // perc
-            max_expansion_time = min(
-                multiplier * (time.time() - spec_search_start), 3600
-            )
-            logger.debug(
-                "Will expand for %s seconds.",
-                round(max_expansion_time, 2),
-            )
+        spec_rules = self._auto_search_rules(**kwargs)
+        if spec_rules is not None:
+            specification = CombinatorialSpecification(self.start_class, spec_rules)
+            self._log_spec_found(specification, auto_search_start)
+            return specification
+        raise SpecificationNotFound
```

Comment:
this can't be reached, since _auto_search_rules raises the same. Should it just be in one place?

Reply:
Unless strong opinion I'll leave it. It's only extra  safety